### PR TITLE
Error messages needs to be corrected

### PR DESCRIPTION
**Pull request by tejaswinichile**
Issue #504

### CODE REVIEW

File: app/javascript/Admin/Main.elm [hunk 1]
```diff
@@ -706,12 +706,38 @@ update msg model =
                         OrganizationCreate.update oCMsg currentPageModel
                 in
                     case oCMsg of
-                        OrganizationCreate.SaveOrgResponse (Ok org) ->
-                            update (NavigateTo (Route.ArticleList org.api_key))
-                                { model
-                                    | organizationKey = org.api_key
-                                    , organizationName = org.name
-                                }
+                        OrganizationCreate.SaveOrgResponse (Ok orgWithErrors) ->
```
Comment (tejaswinichile) [inline app/javascript/Admin/Main.elm]: @ritesh404 Can we do it in a more sophisticated way. For every call, I need to check if errors present in the saveResponse.
I have done this for only one mutation right now. We will be doing this for all mutation request, I think we need to check some refactoring for this now itself.

Comment (tejaswinichile) [reply app/javascript/Admin/Main.elm]: @ritesh404 _a

Comment (ritesh404) [reply app/javascript/Admin/Main.elm]: Looking into it

**Comment by ritesh404** [comment]
@tejaswinichile _a pleas have a look at https://github.com/bigbinary/acehelp/pull/528/commits/7c2625d8fb2a7fc3bc06227ec178e6a278cf4448
I changed the field `error: Maybe String` to `errors: List String` and created a helper function `flattenErrors` to convert graphql errors to `List String` type

**Comment by tejaswinichile** [comment]
@ritesh404 _a Please check I will send more PRs to fix error messages for all other pages.

**Comment by tejaswinichile** [comment]
@ritesh404 _a Please check. I have updated type alias for `OrganizationWithError` and `ArticleWithError`.

File: app/javascript/Admin/Main.elm [hunk 2]
```diff
@@ -361,9 +362,22 @@ update msg model =
                     )
             in
             case caMsg of
-                ArticleCreate.SaveArticleResponse (Ok id) ->
-                    updateNavigation (Route.ArticleList model.organizationKey) ( newModel, newCmds )
-                        |> renderFlashMessages (UserNotification.SuccessNotification "Article created successfully.")
+                ArticleCreate.SaveArticleResponse (Ok articleWithErrors) ->
```
Comment (ritesh404) [inline app/javascript/Admin/Main.elm]: articleWithErrors to articleResponse

File: app/javascript/Admin/Page/Article/Create.elm [hunk 3]
```diff
@@ -92,7 +92,7 @@ update msg model =
             )
 
         SaveArticleResponse (Err error) ->
-            ( { model | error = Just "There was an error while saving the Article", status = None }, [] )
+            ( { model | errors = [] }, [] )
```
Comment (ritesh404) [inline app/javascript/Admin/Page/Article/Create.elm]: Shouldn't we be showing an error here?

File: app/javascript/Admin/Page/Article/Create.elm [hunk 4]
```diff
@@ -103,7 +103,7 @@ update msg model =
                     ( model, [] )
 
         CategoriesLoaded (Err err) ->
-            ( { model | error = Just "There was an error loading Categories" }, [] )
+            ( { model | errors = [] }, [] )
```
Comment (ritesh404) [inline app/javascript/Admin/Page/Article/Create.elm]: Shouldn't we be showing an error here?

File: app/javascript/Admin/Page/Article/List.elm [hunk 5]
```diff
@@ -116,7 +116,7 @@ update msg model =
             ( { model | articles = articles }, [] )
 
         UpdateArticleStatusResponse (Err error) ->
-            ( { model | error = Just "There was an error wile updating the Status" }, [] )
+            ( { model | error = Just "There was an error loading articles" }, [] )
```
Comment (ritesh404) [inline app/javascript/Admin/Page/Article/List.elm]: The error messages appear to be swapped for `UpdateArticleStatusResponse` and  `ArticleListLoaded` why?. also, there is a typo in the message
"There was an error `while` updating the Status"

File: app/javascript/Admin/Page/Article/Create.elm (hunk 3 above)
Comment (tejaswinichile) [reply app/javascript/Admin/Page/Article/Create.elm]: We are handling save article response in `admin/main.elm`

File: app/javascript/Admin/Page/Article/List.elm (hunk 5 above)
Comment (tejaswinichile) [reply app/javascript/Admin/Page/Article/List.elm]: It has been changed while rebasing. Let me check that again.

File: app/javascript/Admin/Page/Article/Create.elm [hunk 6]
```diff
@@ -103,7 +103,7 @@ update msg model =
                     ( model, [] )
 
         CategoriesLoaded (Err err) ->
-            ( { model | error = Just "There was an error loading Categories" }, [] )
+            ( { model | errors = [ "There was an error while saving the Article" ] }, [] )
```
Comment (ritesh404) [inline app/javascript/Admin/Page/Article/Create.elm]: the message should be "There was an error while loading Categories"

**Comment by ritesh404** [comment]
@tejaswinichile _a please resolve conflicts and merge the PR

**Comment by tejaswinichile** [comment]
Resolved conflicts. Merging this.

**Pull request merged**